### PR TITLE
changes from 2/20/24

### PR DESCRIPTION
**Pull request by danielyowell**

### CODE REVIEW

File: lib/buffer.txt [hunk 1]
```diff
@@ -0,0 +1,216 @@
+class PersonalForm extends StatefulWidget {
```
Comment (campcoxconsultants) [inline lib/buffer.txt]: remove this file

File: lib/form_box.dart [hunk 2]
```diff
@@ -15,6 +18,9 @@ class FormBox extends StatefulWidget {
   final String hintText;
   final TextEditingController? controller;
 
+  final TextInputType keyboardType;
+  //final TextInputFormatter inputFormatters;
```
Comment (campcoxconsultants) [inline lib/form_box.dart]: add a todo

File: lib/form_example.dart [hunk 3]
```diff
@@ -136,16 +141,20 @@ class _SignUpFormState extends State<SignUpForm> {
           }
           print(
               'sign up with password: "${_passwordController.text}"/"${_confirmPasswordController.text}"');
+          Navigator.push(context, MaterialPageRoute(builder: (_) {
+//!!!!!!!!!!!! Use this button for testing different screens (default: PersonalForm)
```
Comment (campcoxconsultants) [inline lib/form_example.dart]: make a todo if necessary

File: lib/form_example.dart [hunk 4]
```diff
@@ -136,16 +141,20 @@ class _SignUpFormState extends State<SignUpForm> {
...
-          authHandler.handleSignInEmail(emailController.text, passwordController.text)
-            .then<void>((User user) {
-              Navigator.pushAndRemoveUntil(
-                context,
-                MaterialPageRoute<void>(builder: (context) => const SplashScreen()),
-                (Route<dynamic> route) => false,
-              );
-            }).catchError((e) => print(e));
-
+//           authHandler.handleSignInEmail(emailController.text, passwordController.text)
```
Comment (campcoxconsultants) [inline lib/form_example.dart]: lets do a rebase

File: lib/form_login.dart [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+              _dontHaveAccountRow,
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget get _createLoginButton => TextButton(
+        onPressed: () {
```
Comment (campcoxconsultants) [inline lib/form_login.dart]: extract this method, place it above the build method

File: lib/form_login.dart [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+          ),
+        ],
+      );
+
+  Widget get _dontHaveAccountRow => Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          const Text(
+            'Don\'t have an account?',
```
Comment (campcoxconsultants) [inline lib/form_login.dart]: `"Don't have an account?"`

Comment (campcoxconsultants) [reply lib/form_login.dart]: `'Don’t ...' `

File: lib/form_personal_details.dart [hunk 7]
```diff
@@ -0,0 +1,200 @@
+//!! This might be deprecated.
+// Delete if still not in use after a while.
```
Comment (campcoxconsultants) [inline lib/form_personal_details.dart]: TODO

File: lib/form_personal_details.dart [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+  late TextEditingController _firstNameController;
+  late TextEditingController _lastNameController;
+  late TextEditingController _dateOfBirthController;
+  late TextEditingController _mobileNumberController;
+
+  String _firstNameError = '';
+  String _lastNameError = '';
+  String _dateOfBirthError = '';
+  String _mobileNumberError = '';
+  String required = 'This field is required';
```
Comment (campcoxconsultants) [inline lib/form_personal_details.dart]: rename

File: lib/form_personal_details.dart [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+  //       errorField = required;
+  //     } else {
+  //       errorField = '';
+  //     }
+  //   });
+  // }
+
+  Widget get _createNextButton => TextButton(
+        onPressed: () {
+          _firstNameController.text.isEmpty ? setState(() { _firstNameError = required; }) : setState(() { _firstNameError = ''; });
```
Comment (campcoxconsultants) [inline lib/form_personal_details.dart]: find how we did this in the login (?) widget

File: lib/form_reset_password.dart [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  @override
+  void dispose() {
+    _emailController.dispose();
+
+    super.dispose();
+  }
+
+  late TextEditingController _emailController;
```
Comment (campcoxconsultants) [inline lib/form_reset_password.dart]: late final

File: lib/main.dart [hunk 11]
```diff
@@ -36,7 +37,8 @@ class MyApp extends StatelessWidget {
     return MaterialApp(
       title: 'Flutter Demo',
       theme: ThemeData(fontFamily: 'Poppins'),
-      home: const SplashScreen(),
+      debugShowCheckedModeBanner: false,
+      home: const Homepage(),
```
Comment (campcoxconsultants) [inline lib/main.dart:37]: don't check this in

File: lib/main.dart [hunk 12]
```diff
@@ -36,7 +37,8 @@ class MyApp extends StatelessWidget {
     return MaterialApp(
       title: 'Flutter Demo',
       theme: ThemeData(fontFamily: 'Poppins'),
-      home: const SplashScreen(),
+      debugShowCheckedModeBanner: false,
```
Comment (campcoxconsultants) [inline lib/main.dart:36]: yes to this

File: lib/main.dart [hunk 13]
```diff
@@ -36,7 +37,8 @@ class MyApp extends StatelessWidget {
     return MaterialApp(
       title: 'Flutter Demo',
```
Comment (campcoxconsultants) [inline lib/main.dart:34]: update this to Patient Inform or whatever you want it to be

**Review by danielyowell — COMMENTED** [review]
approved

**Pull request closed without merging**